server/src/main/java/org/elasticsearch/cluster/SnapshotsInProgress.java

```diff
@@ -96,18 +99,52 @@ public static Entry startedEntry(Snapshot snapshot, boolean includeGlobalState,
                 indices, dataStreams, startTime, repositoryStateId, shards, null, userMetadata, version);
     }
 
+    /**
+     * Creates the initial snapshot clone entry
+     *
+     * @param snapshot snapshot to clone into
+     * @param source   snapshot to clone from
+     * @param indices  indices to clone
+     * @param startTime start time
+     * @param repositoryStateId repository state id that this clone is based on
+     * @param version repository metadata version to write
+     * @return snapshot clone entry
+     */
+    public static Entry startClone(Snapshot snapshot, SnapshotId source, List<IndexId> indices, long startTime,
+                                   long repositoryStateId, Version version) {
+        return new SnapshotsInProgress.Entry(snapshot, false, false, State.STARTED, indices, Collections.emptyList(),
```

Comment:
My understanding would be yes since the motivation here was cloning indices. That said, I could just add the ability to clone global meta as well in a follow up? It's not hard to add but adds yet another step of complexity to snapshot finalization so that going into a separate step might be a good idea if we want it?

Reply:
Again I'm afraid of silent loses. If a snapshot contains the global metadata I'd expect its clone to also contains it (or vice versa). I'm ok for a follow up as long as it lands in the feature before it's backported.

Reply:
Actually ... turns out bdd6943f913 is all that's needed to get the global metadata into the clone. The more interesting question is data streams. I'm not sure we would just want to clone those as well every time because of how they interact with indices. That might technically need smarter logic there, I'll look into those in a follow-up.